src/sidebar/components/hooks/use-element-should-close.js

```diff
@@ -4,14 +4,28 @@ import { normalizeKeyName } from '../../../shared/browser-compatibility-utils';
 import { listen } from '../../util/dom';
 
 /**
- * @typedef Ref
+ * @template T
+ * @typedef {import("preact/hooks").Ref<T>} Ref
+ */
+
+/**
+ * @typedef PreactElement
+ * @prop {Node} base
+ */
+
+/**
+ * @typedef {Ref<HTMLButtonElement> & Ref<PreactElement>} PreactRef
+ */
+
+/**
+ * @typedef Ref2
  * @prop current {Node} - HTML node
```

Comment:
It would be good to find a way to reduce the number of different "ref" types here.

Reply:
Sorry, this is leftover cruft from my draft, I will clean this up.